cmake/Modules/FindLibClang.cmake

```diff
@@ -53,14 +54,14 @@ if(NOT LIBCLANG_LDFLAGS)
     # -rpath is relative path for all linked libraries.
     # The second "." is argument to rpath.
 
-    if(UNIX)
-        set(LIBCLANG_LDFLAGS_OS "-L--enable-new-dtags -L-rpath=.")
-    elseif(APPLE)
-        set(LIBCLANG_LDFLAGS_OS "-L-rpath=${LIBCLANG_DIR_PATH}")
+    if(UNIX AND NOT APPLE)
+        set(LIBCLANG_LDFLAGS_OS "-L--enable-new-dtags -L-rpath=. -L--no-as-needed -L-l:${LIBCLANG_LIB}")
+    elseif(UNIX AND APPLE)
+        set(LIBCLANG_LDFLAGS_OS "-L-rpath -L${LIBCLANG_DIR_PATH} -L-lclang")
```

Comment:
The -L${LIBCLANG_DIR_PATH} is redundant, already covered by the flag set at line 64